### PR TITLE
fix: avoid erroring out with simpson rule and n_points < 3

### PR DESCRIPTION
**Pull request by pcolaianni**
Should solve #143

### CODE REVIEW

File: src/openms/source/ANALYSIS/OPENSWATH/PeakIntegrator.cpp [hunk 1]
```diff
@@ -145,21 +168,26 @@ namespace OpenMS
...
-      for (auto it = p.PosBegin(left); it != p.PosEnd(right) - 1; ++it)
-      {
-        peak_area += ((it + 1)->getPos() - it->getPos()) * ((it->getIntensity() + (it + 1)->getIntensity()) / 2.0);
-      }
+      peak_area = compute_peak_area_trapezoid(left, right);
     }
     else if (integration_type_ == INTEGRATION_TYPE_SIMPSON)
     {
-      if (n_points < 3)
+      if (n_points == 1)
```
Comment (dmccloskey) [inline src/openms/source/ANALYSIS/OPENSWATH/PeakIntegrator.cpp:187]: Is there a similar check for Trapezoid integration?

File: src/tests/class_tests/openms/source/PeakIntegrator_test.cpp [hunk 2]
```diff
@@ -513,11 +525,23 @@ START_SECTION(PeakArea integratePeak(
...
     TEST_REAL_SIMILAR(pa.hull_points[i][1], it->getIntensity())
   }
+
+  STATUS("Integration type: simpson (1 point)")
+  pa = ptr->integratePeak(chromatogram, left, 2.478);
+  TEST_REAL_SIMILAR(pa.area, 881)
+  TEST_REAL_SIMILAR(pa.height, 881)
+  TEST_REAL_SIMILAR(pa.apex_pos, 2.477966667)
+
+  STATUS("Integration type: simpson (2 points)")
```
Comment (dmccloskey) [inline src/tests/class_tests/openms/source/PeakIntegrator_test.cpp:556]: Let's add a test (once implemented) to asl check for the case of "Integration type: trapezoid (1 point)"

**Review by dmccloskey — CHANGES_REQUESTED** [review]
Let's also add in a fall back for the trapezoid method with only 1 point.  Simpson looks good.

File: src/openms/source/ANALYSIS/OPENSWATH/PeakIntegrator.cpp (hunk 1 above)
Comment (pcolaianni) [reply src/openms/source/ANALYSIS/OPENSWATH/PeakIntegrator.cpp:187]: No, adding it :+1:

**Pull request closed without merging**